### PR TITLE
Introduce AllowedSecondFactorListUpdatedEvent

### PR DESCRIPTION
**Pull request by arothuis**
[119815553](https://www.pivotaltracker.com/story/show/119815553)

### CODE REVIEW

File: src/Surfnet/Stepup/Configuration/Event/AllowedSecondFactorListUpdatedEvent.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            AllowedSecondFactorList::deserialize($data['allowed_second_factor_list'])
+        );
+    }
+
+    public function serialize()
+    {
+        return [
+            'institution_configuration_id' => $this->institutionConfigurationId->getInstitutionConfigurationId(),
+            'institution'                  => $this->institution->getInstitution(),
+            'allowed_second_factor_list'  => $this->allowedSecondFactorList->serialize(),
```
Comment (DRvanR) [inline src/Surfnet/Stepup/Configuration/Event/AllowedSecondFactorListUpdatedEvent.php]: alignment :stuck_out_tongue:

File: src/Surfnet/Stepup/Configuration/Value/AllowedSecondFactorList.php [hunk 2]
```diff
@@ -86,6 +87,24 @@ public function contains(SecondFactorType $secondFactor)
...
+        $secondFactorTypes = array_map(function ($secondFactorString) {
+            return new SecondFactorType($secondFactorString);
+        }, $data['allowed_second_factors']);
+
+        return new self($secondFactorTypes);
+    }
+
+    public function serialize()
+    {
+        return [
```
Comment (DRvanR) [inline src/Surfnet/Stepup/Configuration/Value/AllowedSecondFactorList.php]: please keep the map out of the return statement :smile:

File: src/Surfnet/Stepup/Tests/Configuration/Event/EventSerializationAndDeserializationTest.php [hunk 3]
```diff
@@ -133,6 +136,16 @@ public function institutionConfigurationEventsProvider()
                     new ShowRaaContactInformationOption(true)
                 )
             ],
+            'AllowedSecondFactorListUpdatedEvent' => [
```
Comment (DRvanR) [inline src/Surfnet/Stepup/Tests/Configuration/Event/EventSerializationAndDeserializationTest.php]: Might want to test with an blank list as well.

File: src/Surfnet/Stepup/Configuration/Value/AllowedSecondFactorList.php [hunk 4]
```diff
@@ -19,11 +19,12 @@
 namespace Surfnet\Stepup\Configuration\Value;
 
 use ArrayIterator;
+use Broadway\Serializer\SerializableInterface;
 use IteratorAggregate;
 use JsonSerializable;
 use Surfnet\StepupBundle\Value\SecondFactorType;
 
-final class AllowedSecondFactorList implements JsonSerializable, IteratorAggregate
+final class AllowedSecondFactorList implements JsonSerializable, IteratorAggregate, SerializableInterface
```
Comment (DRvanR) [inline src/Surfnet/Stepup/Configuration/Value/AllowedSecondFactorList.php]: Just a thought: personally, I prefer to have interface names listed by alphabet. Really is something minor, but I've noticed that mental overhead is lower and that code when adding an interface provides cleaner diffs. Plus expectation on what is listed where etc.  Really can't think of a single truly compelling argument, mostly preference and some minor advantages.

Comment (arothuis) [inline src/Surfnet/Stepup/Configuration/Value/AllowedSecondFactorList.php]: I don't share this preference: it's OK for me either way. Will try to look into it next time.

**Pull request merged**